yt_dlp/extractor/archiveorg.py

```diff
@@ -241,3 +256,153 @@ def _real_extract(self, url):
+            'info_dict': {
+                'id': 'AkhihxRKcrs',
+                'ext': 'webm',
+                'title': 'Limited Run: Mondo\'s Modern Classic 1 of 3 (SDCC 2012)'
+            }
+        },
+        {
+            # Old flash-only video. Webpage title starts with "YouTube - ".
+            'url': 'https://web.archive.org/web/20081211103536/http://www.youtube.com/watch?v=jNQXAC9IVRw',
+            'info_dict': {
+                'id': 'jNQXAC9IVRw',
+                'ext': 'unknown_video',
+                'title': 'Me at the zoo'
+            }
+        },
+        {
+            # Flash video with .flv extension (itag 34). Title has prefix "YouTube         -"
+            # Title has some weird unicode characters too.
+            'url': 'https://web.archive.org/web/20110712231407/http://www.youtube.com/watch?v=lTx3G6h2xyA',
+            'info_dict': {
+                'id': 'lTx3G6h2xyA',
+                'ext': 'flv',
+                'title': '‪Madeon - Pop Culture (live mashup)‬‏'
+            }
+        },
+        {   # Some versions of Youtube have have "Youtube" as page title in html (and later rewritten by js).
+            'url': 'https://web.archive.org/web/http://www.youtube.com/watch?v=kH-G_aIBlFw',
+            'info_dict': {
+                'id': 'kH-G_aIBlFw',
+                'ext': 'mp4',
+                'title': 'kH-G_aIBlFw'
+            },
+            'expected_warnings': [
+                'unable to extract title',
+            ]
+        },
+        {
+            # First capture is a 302 redirect intermediary page.
+            'url': 'https://web.archive.org/web/20050214000000/http://www.youtube.com/watch?v=0altSZ96U4M',
+            'info_dict': {
+                'id': '0altSZ96U4M',
+                'ext': 'mp4',
+                'title': '0altSZ96U4M'
+            },
+            'expected_warnings': [
+                'unable to extract title',
+            ]
+        },
+        {
+            # Video not archived, only capture is unavailable video page
+            'url': 'https://web.archive.org/web/20210530071008/https://www.youtube.com/watch?v=lHJTf93HL1s&spfreload=10',
+            'only_matching': True,
+        }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        title = video_id  # if we are not able get a title
+
+        def _extract_title(webpage):
+            page_title = strip_or_none(self._html_search_regex(r'<title>([^<]*)</title>', webpage, 'title', fatal=False), default='')
+            # YouTube video pages appear to always have either 'YouTube -' as suffix or '- YouTube' as prefix.
+            try:
+                page_title = strip_or_none(self._search_regex(r'(?:YouTube\s*-\s*(.*)$)|(?:(.*)\s*-\s*YouTube$)', page_title, 'title', default=''))
+            except RegexNotFoundError:
+                self.report_warning('unable to extract title')
+                return
```

Comment:
use `fatal=False` in `_search_regex`. Also, if prefix/suffix is not found, return the title as-is. Why fall back t id in that case?
Also, you may want to use `_html_search_regex` instead of `strip_or_none(self._search_regex`

Reply:
If the prefix/suffix is not found, the page title is likely to not be the video title, hence we do not want it. e.g. some pages have placeholder title "YouTube", or the 302 page has empty string.

Suppose I could change it to re.match

Reply:
I think I'll keep it as `_html_search_regex` with `fatal=True`. Since it only raises RegexNotFoundError when fatal, it should be fine. writing own logic will end up rewriting something similar anyways